### ISSUE TITLE
Binding to both IPv4 and IPv6 address

### ISSUE THREAD

**Issue by utrenkner** [skeeto/endlessh#13]
Nice idea - thanks for endlessh!

I just installed it in a FreeBSD jail. Unfortunately, it listens only on the IPv6 address. Is there a way of letting it listen also on the IPv4 address of the jail?

**Comment by borg333** [comment]
Yes, the same, only tcp6
```
# netstat -an | grep 2222
tcp6       0      0 *.2222                 *.*                    LISTEN
```

**Comment by skeeto** [comment]
This is where I'm out of my depth. Originally I only used IPv4 since 
that's what I know best. Then I extended it to IPv6 in 09bb8be. I saw it 
was automatically binding to both IPv4 and IPv6, so I left it at that.

I had done a little research and learned about the IPV6_V6ONLY socket 
option. Fortunately it seems that, when it exists, it's generally 
disabled by default. Your jail must have it enabled by default.

**Comment by borg333** [comment]
I run it on VM esxi on fresh FreeBSD11.2. So, what i must do?

**Comment by PyRo1509** [comment]
This is affecting me as well on a fresh Ubuntu 16 install.  Once this issue is solved I'd love to run this.

**Comment by Inrin** [comment]
This has to do with IPv4 mapped IPv6 address communication.
Linux defaults to allow this, whereas FreeBSD defaults to not allow this and defaults to IPv6Only.

To allow this under Linux use:
```
sysctl -w net.ipv6.bindv6only=0
```
To make it permanent safe it in a `/etc/sysctl.d/40-ipv6only.conf
```
net.ipv6.bindv6only = 0
```

For FreeBSD:
Add:
```
ipv6_ipv4mapping="YES"
```
to your `/etc/rc.conf`

Maybe I'll get the time to make an PR to include -4 -6 command switches.

PS:
@PyRo1509
Why do you use Ubuntu 16.XX in 2019?

Sources:
https://www.freebsd.org/doc/handbook/network-ipv6.html
http://man7.org/linux/man-pages/man7/ipv6.7.html

**Comment by PyRo1509** [comment]
> Why do you use Ubuntu 16.XX in 2019?

You use what that your VPS provider provides.

**Comment by utrenkner** [comment]
@Inrin Thanks for the info. I tried setting ``ipv6_ipv4mapping="YES"`` in my jail, but that does not have any effect. I would probably have to set it on the the host - but I am not going to do this on a production server with several other jails.

**Comment by Inrin** [comment]
@utrenkner
Good to know! I couldn't test it myself as currrently I have no *BSD setup running.
I found https://superuser.com/questions/903308/forcing-bsd-to-accept-both-ipv4-and-ipv6-traffic-on-a-single-socket/999182, which doesn't sound promising.

I'm currently testing adding `-4 -6` switches and default to IPv4 mapped IPv6.
According to https://www.freebsd.org/cgi/man.cgi?query=ip6&apropos=0&sektion=0&manpath=FreeBSD+12.0-RELEASE+and+Ports&arch=default&format=html FreeBSD also has the option, so it sounds promising. (Need to spawn an FreeBSD instance for testing though)

**Comment by skeeto** [comment]
This should now be resolved by #20. Thanks, @Inrin!
